### PR TITLE
Introduce screenreader-only internal component

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

A follow-up for https://github.com/cloudscape-design/components/pull/351.

### How has this been tested?

Manual testing.

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [x] _No._

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/components/screenreader-only/index.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import React from 'react';
+import styles from './styles.css.js';
+
+type ScreenreaderOnlyProps = Exclude<React.HTMLAttributes<HTMLDivElement>, 'className'>;
+
+/**
+ * Makes content now shown on a screen but still announced by screen-reader users.
+ * The component is suitable when the aria-label cannot be used, e.g. to avoid elemnts being announced as "blank".
+ *
+ * To exclude screenreader-only content for testing use `wrapper.find(':not([data-screenreader-only])')`.
```
Comment (just-boris) [inline src/internal/components/screenreader-only/index.tsx:13]: Not sure how to use this guideline, could you give a more detailed example?

Comment (pan-kot) [reply src/internal/components/screenreader-only/index.tsx:13]: Example: https://github.com/cloudscape-design/components/pull/351/files#diff-5f254303359ce4f81256cf4e5809f2736eba9f5262de9047982f02b8e7864a82R60

Comment (just-boris) [reply src/internal/components/screenreader-only/index.tsx]: There is a class name directly, which can be used

```js
import visuallyHiddenStyles from '~internal/components/screenreader-only/styles.css.js'

wrapper.findAll(':not(${visuallyHiddenStyles.root})')
```

The tactic is always the same – do not add extra data-attributes unless absolutely necessary

Comment (pan-kot) [reply src/internal/components/screenreader-only/index.tsx]: Alright, I don't mind it.

**Review by just-boris — APPROVED** [review]

**Pull request merged**